crates/vm/src/arch/execution.rs

```diff
@@ -107,6 +109,7 @@ pub trait InstructionExecutor<F> {
         &mut self,
         memory: &mut MemoryController<F>,
         streams: &mut Streams<F>,
+        rng: &mut StdRng,
```

Comment:
do we really need this in the interface of the trait? it's not great because the `rng` is only used so sparingly

Reply:
discussed offline; we'll soon change the traits to take a `VmStateMut` to hide the state away